In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt  

In [2]:
results_cpu = pd.read_csv('../benchmarks/random_forest/results11_cpu.csv')[1:]
results_tpu = pd.read_csv('../benchmarks/random_forest/results11_edgetpu.csv')[1:]

results_less_tpu_if = pd.read_csv('../benchmarks/random_forest/results11_less_tpu.csv')[1:]

results_1opp = pd.read_csv('../benchmarks/random_forest/results_1opp.csv')[1:]
results_2opp = pd.read_csv('../benchmarks/random_forest/results_2opp.csv')[1:]
results_3opp = pd.read_csv('../benchmarks/random_forest/results_3opp.csv')[1:]
results_4opp = pd.read_csv('../benchmarks/random_forest/results_4opp.csv')[1:]

In [5]:
mean_cpu = results_cpu['TimeMS'].mean()
mean_tpu = results_tpu['TimeMS'].mean()

mean_less_tpu_if = results_less_tpu_if['TimeMS'].mean()

mean_1opp = results_1opp['TimeMS'].mean()
mean_2opp = results_2opp['TimeMS'].mean()
mean_3opp = results_3opp['TimeMS'].mean()
mean_4opp = results_4opp['TimeMS'].mean()

In [7]:
print(f'Mean CPU:\t\t{mean_cpu : .2f}ms\nMean TPU:\t\t{mean_tpu : .2f}ms')
print(f'~ Time for tpu opps:\t{mean_less_tpu_if : .2f}ms')
print(f'Mean one opp:\t\t{mean_1opp : .2f}ms')
print(f'Mean two opp:\t\t{mean_2opp : .2f}ms')
print(f'Mean three opp:\t\t{mean_3opp : .2f}ms')
print(f'Mean four opp:\t\t{mean_4opp : .2f}ms')

Mean CPU:		 0.03ms
Mean TPU:		 1.04ms
~ Time for tpu opps:	 0.53ms
Mean one opp:		 0.52ms
Mean two opp:		 0.47ms
Mean three opp:		 0.53ms
Mean four opp:		 0.53ms


### Ergebnis
Die volle Inferenz auf der CPU dauert $0.03$ ms. <br>
TPU mit If-Abfolge benötigt $1.04$ ms. <br>
Das Entfernen der $8$ CPU Operationen ermöglicht das volle Mappen auf die Edge TPU. <br> 
Die $8$ Operationen auf der Edge TPU benötigen $0.53$ ms. <br> 
Also benötigt die Kommunikation zwischen TPU und CPU in etwa $0.51$ ms.